sparkly/reader.py

```diff
@@ -214,6 +219,71 @@ def mysql(self, host, database, table, port=None, parallelism=None, options=None
 
         return self._basic_read(reader_options, options, parallelism)
 
+    def kafka(self,
+              brokers,
+              topics=None,
+              offset_ranges=None,
+              key_deserializer=None,
+              value_deserializer=None,
+              schema=None,
+              parallelism=None,
+              options=None):
+        """Creates dataframe from specified set of messages from Kafka topic.
+
+        If `offset_ranges` parameter is omitted it will auto-discover.
+        If `schema` parameter is specified the Dataframe with this schema will be returned,
+        else the RDD will be returned.
+        The `schema` parameter, if specified, should contain two top level fields:
+        `key` and `value`.
+
+        Args:
+            brokers (list): List of kafka brokers.
+            topics (list[str]): List of kafka topics to read from.
+            offset_ranges (list[(str, int, int, int)]): List of partition ranges
+                [(topic, partition, start_offset, end_offset)].
+            key_deserializer (function): Function used to deserialize the key.
+            value_deserializer (function): Function used to deserialize the value.
+            schema (pyspark.sql.types.StructType|None): Schema to apply to create a Dataframe.
```

Comment:
what is the type of data that key/value serialisers return? Can we infer schema from it?